alibi_detect/cd/base.py

```diff
@@ -268,6 +292,50 @@ def predict(self, x: Union[np.ndarray, list], return_p_val: bool = True,
             cd['data']['model'] = self.model
         return cd
 
+    @abstractmethod
+    def get_config(self) -> dict:
+        """
+        Get the detector's configuration dictionary.
+
+        Returns
+        -------
+        The detector's configuration dictionary.
+        """
+        cfg: Dict[str, Any] = {
+            'version': __version__,
+            'config_spec': __config_spec__,
+            'name': 'ClassifierDrift'
+        }
+
+        # x_ref
+        cfg.update({'x_ref': self.x_ref})
+
+        # Preprocess field
+        if self.preprocess_fn is not None:
+            cfg.update({'preprocess_fn': self.preprocess_fn})
+
+        # Detector kwargs
+        # Note: Currently write kwargs out such that load_detector will load a fresh detector. We don't (yet) write out
+        #  stateful attributes such as self.skf, which would enable the detector to be re-loaded without re-fitting.
+        kwargs = {
+            'p_val': self.p_val,
+            'x_ref_preprocessed': self.preprocess_at_init or self.x_ref_preprocessed,
+            'preprocess_at_init': self.preprocess_at_init,
+            'update_x_ref': self.update_x_ref,
+            'preprocess_fn': self.preprocess_fn,
+            'preds_type': self.preds_type,
+            'binarize_preds': self.binarize_preds,
+            'train_size': self.train_size,
+            'n_folds': self.skf.n_splits if self.skf is not None else None,
```

Comment:
Some kwargs are conditional on others, and kwarg's are not always added explicitly as class attributes. Therefore this sort of logic is required in `get_config`'s to infer kwarg's from attributes.